ossdev/__init__.py

```diff
@@ -21,51 +25,183 @@ def set(self, arr):
     def get(self):
         return self.d
 
+    def clone(self):
+        return Vector.from_arr(list(self.get()))
+
+    @property
+    def is_row(self):
+        return self.row
+
     def __len__(self):
         return len(self.d)
 
     def __repr__(self):
         return str(self.d)
 
+    def __str__(self):
+        return 'Vct%s' % self.d
+
     def __getitem__(self, item):
         return self.d[item]
 
     def __hash__(self):
         return sum(self.d)
 
     def __setitem__(self, key, value):
-        # TODO: implement
-        return None
+        self.d[key] = value
 
     def __cmp__(self, other):
-        # TODO: implement, -1 if self < other, 0 if self == other, 1 if self > other
-        return -1
+        return other is not None and self.length() - other.length()
+
+    def __eq__(self, other):
```

Comment:
Why is the comparison done in this way? `__cmp__` substitutes all other magic comparison methods.
Moreover, it is quite similar to this https://github.com/crocs-muni/open-source-development-course-hw02-1/pull/4/files  why?

Reply:
At first I tried implementing only __cmp__ but when I tested comparison it gave me an error
```
python3 -i ossdev/__init__.py
>>> a = Vector()
>>> b = Vector()
>>> a < b 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<' not supported between instances of 'Vector' and 'Vector'
```
Then I tried running the same code with python 2, which worked.
```
$ python2 -i ossdev/__init__.py
>>> a = Vector()
>>> b = Vector()
>>> a < b 
False
```
I searched the documentation you linked us to (https://rszalski.github.io/magicmethods/) for `__cmp__` and I found:
> # Appendix 2: Changes in Python 3
> ...
> - __cmp__ is gone due to redundancy with other magic methods

So I thought that it is necessary to implement the other functions (as the library info stated required version is 3).

Reply:
Ok cool. Next time you can use https://docs.python.org/3/library/functools.html#functools.total_ordering to simplify the task.

Still, comparing by vector length only is not enough. However, we wont fix it now.
Consider this closed. Thanks.